lib/file.js

```diff
@@ -42,3 +43,26 @@ exports.writeToFile = function (path, data) {
     }
   });
 };
+
+/**
+ * Normalize the input/output file path according to a sub-directory if needed
+ * @param {String} file - path of the input/ouput file
+ * @param {String|null} subDirectory - an explicit path to a sub-directory if given
+ * @returns {Promise}
+ */
+exports.normalizeFilePath = function (file, subDirectory) {
+  return Bluebird.resolve()
+  .then(function () {
+    if (!subDirectory) {
+      return file;
+    }
+
+    file = Path.normalize(file);
+
+    if (Path.dirname(file) === '.') {
```

Comment:
Sorry for not replying on the issue earlier, but I think the desired behavior of how `-f` and `-s` will work together will be to always have it be relative from the subdirectory. I think that will give the most options for where the file should go. For example, in the current implementation, I'm not sure there's any way to write the file to root directory of the repo (I tried `CHANGELOG.md`, `../CHANGELOG.md`, `./../CHANGELOG.md`).

And since this is a completely new feature, we don't have to worry about backwards compatibility as long as we don't affect the current functionality. So we should pick what makes the most sense and provides the greatest control.